packages/kbn-legacy-logging/src/log_format_json.test.ts

```diff
@@ -39,30 +39,45 @@ describe('KbnLoggerJsonFormat', () => {
       expect(message).toBe('undefined');
     });
 
-    it('response', async () => {
-      const event = {
-        ...makeEvent('response'),
-        statusCode: 200,
-        contentLength: 800,
-        responseTime: 12000,
-        method: 'GET',
-        path: '/path/to/resource',
-        responsePayload: '1234567879890',
-        source: {
-          remoteAddress: '127.0.0.1',
-          userAgent: 'Test Thing',
-          referer: 'elastic.co',
-        },
-      };
-      const result = await createPromiseFromStreams<string>([createListStream([event]), format]);
-      const { type, method, statusCode, message, req } = JSON.parse(result);
-
-      expect(type).toBe('response');
-      expect(method).toBe('GET');
-      expect(statusCode).toBe(200);
-      expect(message).toBe('GET /path/to/resource 200 12000ms - 13.0B');
-      expect(req.remoteAddress).toBe('127.0.0.1');
-      expect(req.userAgent).toBe('Test Thing');
+    describe('response', () => {
+      it('handles a response object', async () => {
+        const event = {
+          ...makeEvent('response'),
+          statusCode: 200,
+          contentLength: 800,
+          responseTime: 12000,
+          method: 'GET',
+          path: '/path/to/resource',
+          responsePayload: '1234567879890',
+          source: {
+            remoteAddress: '127.0.0.1',
+            userAgent: 'Test Thing',
+            referer: 'elastic.co',
+          },
+        };
+        const result = await createPromiseFromStreams<string>([createListStream([event]), format]);
+        const { type, method, statusCode, message, req } = JSON.parse(result);
+
+        expect(type).toBe('response');
+        expect(method).toBe('GET');
+        expect(statusCode).toBe(200);
+        expect(message).toBe('GET /path/to/resource 200 12000ms - 13.0B');
+        expect(req.remoteAddress).toBe('127.0.0.1');
+        expect(req.userAgent).toBe('Test Thing');
+      });
+
+      it('leaves payload size empty if not available', async () => {
```

Comment:
Leaving entries out of log messages when they aren't defined makes sense but I think we need to explicitly document which fields may or may not be present and why.
Folks might run into issues when, for example, they're ingesting logs by parsing the log message only (as apposed to ingesting the log meta data), and will need to safeguard against potentially missing data. I'm not sure where the best place would be to document optional fields though and if I'm being paranoid here.
WDYT?

Reply:
> I think we need to explicitly document which fields may or may not be present and why.

++ This makes sense to me. I can add the expected fields to the README similar to what you did in https://github.com/elastic/kibana/pull/88070, and call out which are optional. That feels like the most natural place to do it.

> Folks might run into issues when, for example, they're ingesting logs by parsing the log message only (as apposed to ingesting the log meta data), and will need to safeguard against potentially missing data

That's a good point, but I think your suggestion of documenting the fields is a fair way to address this. Even if someone is using something like grok filters to pattern match fields in a message string, they'll still have the ability to mark fields as optional - so it's just a matter of making it clear which ones aren't guaranteed to be present.

Since the `message` field is generally intended to be human-readable, I tend to prefer this approach over figuring out some placeholder value to include for the "missing" fields.